odoo/addons/base/tests/test_expression.py

```diff
@@ -1790,6 +1790,121 @@ def test_rec_names_search(self):
+        ''']):
+            res_company = self.env['res.company']._where_calc([])
+            self.env.execute_query(res_company.select(
+                res_company._.child_ids.name,
+            ))
+
+        with self.assertQueries(['''
+             SELECT "res_company__child_ids"."name"
+               FROM "res_company"
+          LEFT JOIN "res_company" AS "res_company__child_ids"
+                 ON ("res_company"."id" = "res_company__child_ids"."parent_id")
+              WHERE (
+                        "res_company"."active" IS TRUE
+                        AND "res_company"."id" IN (
+                            SELECT "res_company"."parent_id"
+                              FROM "res_company"
+                             WHERE (
+                                       "res_company"."active" IS TRUE
+                                       AND "res_company"."parent_id" IS NOT NULL
+                                   )
+                        )
+                    )
+        ''']):
+            res_company = self.env['res.company']._where_calc([
+                ('child_ids.active', '!=', False),
+            ])
+            self.env.execute_query(res_company.select(
+                res_company._.child_ids.name,
+            ))
+
+    def test_query_builder_many2many(self):
+        with self.assertQueries(['''
+            SELECT "res_company__user_ids__user_ids__partner_id"."name"
+              FROM "res_company"
+         LEFT JOIN "res_company_users_rel" AS "res_company__user_ids"
+                ON ("res_company"."id" = "res_company__user_ids"."cid")
+         LEFT JOIN "res_users" AS "res_company__user_ids__user_ids"
+                ON ("res_company__user_ids"."user_id" = "res_company__user_ids__user_ids"."id")
+         LEFT JOIN "res_partner" AS "res_company__user_ids__user_ids__partner_id"
+                ON ("res_company__user_ids__user_ids"."partner_id" = "res_company__user_ids__user_ids__partner_id"."id")
+             WHERE "res_company"."active" IS TRUE
+
+        ''']):
+            res_company = self.env['res.company']._where_calc([])
+            self.env.execute_query(res_company.select(
+                res_company._.user_ids.name,
+            ))
+
+
+    def test_query_builder_orderby(self):
+        with self.assertQueries(['''
+            SELECT "res_company"."id"
+              FROM "res_company"
+         LEFT JOIN "res_partner" AS "res_company__partner_id"
+                ON ("res_company"."partner_id" = "res_company__partner_id"."id")
+             WHERE "res_company"."active" IS TRUE
+          ORDER BY "res_company__partner_id"."name",
+                   "res_company__partner_id"."id"
+        ''']):
+            res_company = self.env['res.company']._where_calc([])
+            res_company.order = [
+                res_company._.partner_id.name,
+                res_company._.partner_id.id
+            ]
+            self.env.execute_query(res_company.select())
```

Comment:
From an API point of view, I find this quite obscure.
I would propose this instead.  Your `SQLable` is essentially an SQL identifier generator that interacts with a model to add the necessary joins to the `Query` object and manages aliases.
* `query.table` returns an `SQLable` equivalent to an SQL table identifier (`"res_company"`)
* `query.table.partner_id` returns an `SQLable` equivalent to a column identifier (`"res_company"."partner_id"`)
* `query.table.partner_id.name` returns an `SQLable` equivalent to a column identifier with a generated join alias (`"res_company__partner_id"."name"`)

And I would rewrite your code as
```suggestion
            query = self.env['res.company']._where_calc([])
            query.order = SQL(
                "%s, %s",
                query.table.partner_id.name,
                query.table.partner_id,
            )
            self.env.execute_query(res_company.select())
```